nix/kexec-installer/kexec-run.sh

```diff
@@ -74,6 +74,9 @@ if ! "$SCRIPT_DIR/kexec" --load "$SCRIPT_DIR/bzImage" \
   exit 1
 fi
 
+sync; echo 3 > /proc/sys/vm/drop_caches
+echo "current available memory: $(free -h | awk '/^Mem/ {print $7}')"
```

Comment:
Doesn't the linux kernel do this automatically anyway if it needs memory.

Reply:
I could very well imagine that yes, although I didn't verify and don't intent to sake that claim either.

The main reason I put this here is for reporting so that (especially during tests) we can see how much effective free memory was available prior to switching the kernel.

Not sure if it is utterly useful otherwise.

Reply:
Ok. But you are already printing the available memory, which will automatically substract any memory claimed by page cache or dirty pages.
So no need to flush the caches, which might even contain the initrd that we are about to kexec into.

Reply:
Here we could even print a warning if we are below X MB available memory.

Reply:
> Here we could even print a warning if we are below X MB available memory.

I had that idea to: outright refuse to proceed because it's potentially not recoverable, if you have no IPMI access.

Reply:
> Ok. But you are already printing the available memory, which will automatically substract any memory claimed by page cache or dirty pages.
So no need to flush the caches, which might even contain the initrd that we are about to kexec into.

Iirc, I did some A/B testing and saw a small ~30MB difference, but yeah, I agree that it shouldn't be necessary.

I did notice, however, on a slightly tangential note, that at the RAM-limit, `kexec --load` kept working while then `kexec -e` failed. But I guess that's just due to what the stage 1 or squashfs ultimately command into RAM.